### PR TITLE
feature: Token Factory Admin Denom WASM Binding

### PR DESCRIPTION
**Pull request by alexanderbez**
- Add new `DenomAdmin` and `DenomAdminResponse` types
- Update `QueryPlugin` to take a token factory keeper
- Implement `QueryPlugin#GetDenomAdmin` (the actual binding)
- Cleanup/lint `QueryPlugin` & `CustomQuerier`

I will update the Rust side after this PR is merged.

closes: #1603

### CODE REVIEW

File: app/wasm/query_plugin.go [hunk 1]
```diff
@@ -30,15 +31,31 @@ func CustomQuerier(osmoKeeper *QueryPlugin) func(ctx sdk.Context, request json.R
...
+
+		case contractQuery.DenomAdmin != nil:
+			res, err := qp.GetDenomAdmin(ctx, contractQuery.DenomAdmin.Subdenom)
+			if err != nil {
+				return nil, err
+			}
+
+			bz, err := json.Marshal(res)
+			if err != nil {
+				return nil, fmt.Errorf("failed to JSON marshal DenomAdminResponse response")
```
Comment (p0mvn) [inline app/wasm/query_plugin.go:50]: Why not return `err`?

**Review by p0mvn — APPROVED** [review]
LGTM, please also add a unit test for the new query

File: app/wasm/query_plugin.go (hunk 1 above)
Comment (alexanderbez) [reply app/wasm/query_plugin.go:50]: I forgot to wrap the err!

**Comment by alexanderbez** [comment]
I'm going to close this once I wrap up the Rust side.

**Pull request merged**